### PR TITLE
Add prettier pre-commit hook to auto format JS

### LINKED ISSUE #117: Add JS auto-formatter to use with pre-commit
https://prettier.io/
https://prettier.io/docs/en/precommit.html#option-3-pre-commithttpsgithubcompre-commitpre-commit
https://prettier.io/docs/en/configuration.html#basic-configuration

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #117

### CODE REVIEW

File: src/qmxgraph/page/api.js [hunk 1]
```diff
@@ -422,33 +470,43 @@ graphs.Api.prototype._insertDecorationOnEdge = function _insertDecorationOnEdge
...
         parent = graph.getDefaultParent();
         coords = graphs.utils.adjustCoordinates(graph, x, y);
     }
 
-    var tableStyle = 'table';
-    if (style != null) {  // jshint ignore:line
-        tableStyle += ';' + style;
+    var tableStyle = "table";
+    if (style != null) {
+        // jshint ignore:line
```
Comment (oliveira-mauricio) [inline src/qmxgraph/page/api.js:501]: Not sure what this was supposed to suppress, but it was moved around.

File: src/qmxgraph/page/api.js [hunk 2]
```diff
@@ -596,25 +654,26 @@ graphs.Api.prototype.getDecorationParentCellId = function getDecorationParentCel
...
         y = geometry.offset ? geometry.offset.y : 0;
     }
 
     return {
-        'x': x,
-        'y': y,
-        'width': geometry.width,
-        'height': geometry.height,
-        'parent_anchor_position': parent_anchor_position
+        x: x,
```
Comment (oliveira-mauricio) [inline src/qmxgraph/page/api.js:672]: I didn't know about this (`x` and `'x'` both work the same here).

Comment (prusse-martin) [reply src/qmxgraph/page/api.js:672]: yes it does

File: src/qmxgraph/page/api.js [hunk 3]
```diff
@@ -1277,96 +1338,100 @@ graphs.Api.prototype.registerTerminalChangedHandler = function registerTerminalC
  * boolean indicating if the changed terminal is the source (or target), id of the new terminal,
  * port id used in the new terminal, id of the old terminal, port id used in the old terminal.
  */
-graphs.Api.prototype.registerTerminalWithPortChangedHandler = function registerTerminalWithPortChangedHandler (handler) {
-    "use strict";
+graphs.Api.prototype.registerTerminalWithPortChangedHandler =
```
Comment (prusse-martin) [inline src/qmxgraph/page/api.js:1341]: =/
https://prettier.io/docs/en/ignore.html#javascript

> A JavaScript comment of // prettier-ignore will exclude the next node in the abstract syntax tree from formatting.

I ask myself if by "node" they mean just a node or the full sub tree.

**Review by prusse-martin — APPROVED** [review]

**Comment by nicoddemus** [comment]
@prusse-martin about your comments: I don't have strong feelings about how JS should be formatted, so if you find `prettier` settings that you prefer, just let me know and I can update the settings and run again on all files.

**Comment by prusse-martin** [comment]
Those where just rants...
And I didn't find option in playground that changed the output way I have some strong inclination about.

**Pull request merged**